src/toil/batchSystems/torque.py

```diff
@@ -31,29 +31,62 @@
 
 
 
+        
+
 class TorqueBatchSystem(AbstractGridEngineBatchSystem):
-    
 
 
     # class-specific Worker
     class Worker(AbstractGridEngineBatchSystem.Worker):
+        def __init__(self):
+            self._version = None
+
+        def _pbsVersion():
+            """ Determines PBS/Torque version via pbsnodes
+            """
+            try:
+                subprocess.check_output(["pbsnodes", "--version"])
+            except subprocess.CalledProcessError as e:
+               if e.returncode != 0:
+                    logger.error("Could not determine PBS/Torque version")
+               elif "PBSPro" in e.output:
+                    return "pro"
+               else:
+                    return "oss"
+        
+        # Determine if we are in PBSPro or OSS variant of Torque/PBS batch system
+        self._version = _pbsVersion()
```

Comment:
Maybe a self caching method (always call `_pbsVersion()`, which caches the result in `self._version`) instead of referring to the class level attribute (`self._version`) as the former will only get called when needed but the later is getting called inappropriately?